src/AaveCurator/AaveCurator.sol

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.19;
+
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {OwnableWithGuardian} from 'solidity-utils/contracts/access-control/OwnableWithGuardian.sol';
+import {Initializable} from 'solidity-utils/contracts/transparent-proxy/Initializable.sol';
+import {SafeERC20} from 'solidity-utils/contracts/oz-common/SafeERC20.sol';
+import {AaveV2Ethereum, AaveV2EthereumAssets} from 'aave-address-book/AaveV2Ethereum.sol';
+import {AaveV3Ethereum, AaveV3EthereumAssets} from 'aave-address-book/AaveV3Ethereum.sol';
+import {AaveGovernanceV2} from 'aave-address-book/AaveGovernanceV2.sol';
+
+import {IPriceChecker} from './interfaces/IExpectedOutCalculator.sol';
+import {IMilkman} from './interfaces/IMilkman.sol';
+
+contract AaveCurator is Initializable, OwnableWithGuardian {
+  using SafeERC20 for IERC20;
+
+  event DepositedIntoV2(address indexed token, uint256 amount);
+  event DepositedIntoV3(address indexed token, uint256 amount);
+  event SwapCanceled(address fromToken, address toToken, uint256 amount);
+  event SwapRequested(address fromToken, address toToken, uint256 amount);
+  event TokenUpdated(address indexed token, bool allowed);
+
+  error Invalid0xAddress();
+  error InvalidAmount();
+  error InvalidRecipient();
+  error InvalidToken();
+  error OracleNotSet();
+
+  address public constant BAL80WETH20 = 0x5c6Ee304399DBdB9C8Ef030aB642B10820DB8F56;
+  address public constant BPT_PRICE_CHECKER = 0xBeA6AAC5bDCe0206A9f909d80a467C93A7D6Da7c;
+
+  address public chainlinkPriceChecker = 0xe80a1C615F75AFF7Ed8F08c9F21f9d00982D666c;
+  address public milkman = 0x11C76AD590ABDFFCD980afEC9ad951B160F02797;
+
+  mapping(address tokenAddress => bool) public allowedFromTokens;
+  mapping(address tokenAddress => bool) public allowedToTokens;
+  /// @notice Chainlink Oracle address for given token (supports only USD bases)
+  mapping(address tokenAddress => address) public tokenChainlinkOracle;
+
+  function initialize() external initializer {
+    _transferOwnership(_msgSender());
+    _updateGuardian(_msgSender());
+  }
+
+  function swap(
+    address fromToken,
+    address toToken,
+    address recipient,
+    uint256 amount,
+    uint256 slippage
+  ) external onlyOwnerOrGuardian {
+    if (amount == 0) revert InvalidAmount();
+    if (!allowedFromTokens[fromToken]) revert InvalidToken();
+    if (!allowedToTokens[toToken]) revert InvalidToken();
+    if (recipient != address(this) && recipient != address(AaveV3Ethereum.COLLECTOR)) {
```

Comment:
I can see this contract being used in conjunction with the StrategicAssetManager #208 so maybe recipient can also be an allowedAddress and we add a mapping. Thoughts?